ExeterTeam25/group-software-project
# Image fixes

Fixing images (pfps and responses)

To test profile pic crop fix:
- Launch server
- Log into account
- Set a profile picture
- Verify it has correctly been cropped into a square

To test response resize:
- Submit response to challenge. Pick a very wide image
- Submit response to another challenge. Pick a very tall image.
- Verify that they have been resized, maintaining aspect ratio
- Max height should be 500
- Max width should be 750

To test response submission fix:
- Launch server
- Log into account
- delete existing profile picture on your drive (media/profile_pictures)
- attempt to submit image response to challenge
- Verify response worked

Details:
Made it so profile pictures are cropped to 1:1 ratio instead of stretched
Made it so responses are resized while maintaining aspect ratio instead of stretched
Fixed an error where the program would try to access "photograph" instead of "picture" in profile
Fixed an error where profile pictures would break challenge response image uploading
Changed challenge pic css to scale height instead of width (better for tall images)
